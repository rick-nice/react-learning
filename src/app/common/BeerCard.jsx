import React from 'react'
import { Card, Icon } from 'antd'
import PropTypes from 'prop-types'

import beerImg from '../images/beer.svg'

const { Meta } = Card

const BeerCard = ({ id, img, name, description, setFavorite }) => {
  const onBtnClick = () => {
    setFavorite(id)
  }
  return (
    <Card
      style={{ width: 300, heigth: 400 }}
      cover={<img alt='beerImg' src={img || beerImg} heigth={400} />}
      actions={[<Icon type='heart' onClick={onBtnClick} />]}
    >
      <Meta title={name} description={description} />
    </Card>
  )
}

BeerCard.defaultProps = {
  img: beerImg,
}

BeerCard.propTypes = {
  id: PropTypes.number.isRequired,
  img: PropTypes.string,
  name: PropTypes.string.isRequired,
  setFavorite: PropTypes.func.isRequired,
  description: PropTypes.string.isRequired,
}

export default BeerCard
