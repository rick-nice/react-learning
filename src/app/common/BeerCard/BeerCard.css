.beer-card-img {
  margin: auto;
  height: 400px;
}