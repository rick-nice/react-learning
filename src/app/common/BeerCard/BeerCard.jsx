import React from 'react'
import { Card, Icon } from 'antd'
import PropTypes from 'prop-types'

import defaultBeerImg from './images/beer.svg'

const { Meta } = Card

const BeerCard = ({ id, img, name, description, setFavorite }) => {
  const onFavoriteClick = () => {
    setFavorite(id)
  }

  return (
    <Card
      style={{ width: 300, heigth: 400 }}
      cover={<img alt='beerImg' src={img || defaultBeerImg} heigth={400} />}
      actions={[<Icon type='heart' onClick={onFavoriteClick} />]}
    >
      <Meta title={name} description={description} />
    </Card>
  )
}

BeerCard.defaultProps = {
  img: defaultBeerImg,
}

BeerCard.propTypes = {
  id: PropTypes.number.isRequired,
  img: PropTypes.string,
  name: PropTypes.string.isRequired,
  setFavorite: PropTypes.func.isRequired,
  description: PropTypes.string.isRequired,
}

export default BeerCard
