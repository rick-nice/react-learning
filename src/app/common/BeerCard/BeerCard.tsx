import React, { MouseEvent } from 'react'
import { Card, Icon } from 'antd'

import defaultBeerImg from './images/beer.svg'

const { Meta } = Card

interface IProps {
  id: number
  img?: string
  name: string
  description: string
  setFavorite: (id: number) => void
  onCardClick: (id: number) => void
}

const BeerCard = ({
  id,
  img,
  name,
  description,
  setFavorite,
  onCardClick,
}: IProps) => {
  const onFavoriteClick = () => {
    setFavorite(id)
  }

  const handleCardClick = (e: MouseEvent) => {
    onCardClick(id)
  }

  return (
    <Card
      style={{ width: 300, height: 450 }}
      cover={
        <img
          alt='beerImg'
          src={img || defaultBeerImg}
          style={{ width: 200, height: 400 }}
        />
      }
      actions={[<Icon key={1} type='heart' onClick={onFavoriteClick} />]}
      onClick={handleCardClick}
    >
      <Meta title={name} description={description} />
    </Card>
  )
}

export default BeerCard
