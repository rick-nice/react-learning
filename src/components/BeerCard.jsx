import React from 'react'
import { Card, Icon } from 'antd'
import PropTypes from 'prop-types'

const { Meta } = Card

const BeerCard = ({ id, img, title, setFavorite }) => {
  const onBtnClick = () => setFavorite(id)
  return (
    <Card
      style={{ width: 300 }}
      cover={<img alt="beerImg" src={img} />}
      actions={<Icon type="heart" onClick={onBtnClick} />}
    >
      <Meta title={title} description="This is the description" />
    </Card>
  )
}

BeerCard.propTypes = {
  id: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  setFavorite: PropTypes.func.isRequired,
}

export default BeerCard
