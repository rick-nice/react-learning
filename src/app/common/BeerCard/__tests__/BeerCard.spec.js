import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'

import BeerCard from '../BeerCard'
import defaultBeerImg from '../images/beer.svg'

Enzyme.configure({ adapter: new Adapter() })

describe('BeerCard component', () => {
  const props = {
    id: '3',
    img: defaultBeerImg,
    name: 'beer',
    description: 'some beer',
    toggleFavorite: () => {},
    onCardClick: () => {},
  }

  it('render correctly card component', () => {
    const BeerCardComponent = renderer.create(<BeerCard {...props} />).toJSON()
    expect(BeerCardComponent).toMatchSnapshot()
  })
})
