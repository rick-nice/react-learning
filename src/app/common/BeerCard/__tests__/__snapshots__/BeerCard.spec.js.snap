// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard component render correctly card component 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="beerImg"
      className="beer-card-img"
      src="beer.svg"
      style={
        Object {
          "width": 200,
        }
      }
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-title"
        >
          beer
        </div>
        <div
          className="ant-card-meta-description"
        >
          some beer
        </div>
      </div>
    </div>
  </div>
  <ul
    className="ant-card-actions"
  >
    <li
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <span>
        <i
          aria-label="icon: heart"
          className="anticon anticon-heart"
          onClick={[Function]}
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="heart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
            />
          </svg>
        </i>
      </span>
    </li>
    <li
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <span>
        <i
          aria-label="icon: select"
          className="anticon anticon-select"
          onClick={[Function]}
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="select"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h360c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H184V184h656v320c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V144c0-17.7-14.3-32-32-32zM653.3 599.4l52.2-52.2a8.01 8.01 0 0 0-4.7-13.6l-179.4-21c-5.1-.6-9.5 3.7-8.9 8.9l21 179.4c.8 6.6 8.9 9.4 13.6 4.7l52.4-52.4 256.2 256.2c3.1 3.1 8.2 3.1 11.3 0l42.4-42.4c3.1-3.1 3.1-8.2 0-11.3L653.3 599.4z"
            />
          </svg>
        </i>
      </span>
    </li>
  </ul>
</div>
`;
